fractal_server/app/runner/__init__.py

```diff
@@ -110,19 +124,37 @@ async def submit_workflow(
     logger.info(f"output_path: {output_path}")
     logger.info(f"input metadata: {input_dataset.meta}")
     logger.info(f"START workflow {workflow.name}")
-    output_dataset.meta = await process_workflow(
-        workflow=workflow,
-        input_paths=input_paths,
-        output_path=output_path,
-        input_metadata=input_dataset.meta,
-        username=username,
-        workflow_dir=WORKFLOW_DIR,
-        logger_name=logger_name,
-        worker_init=worker_init,
-    )
+    job.working_dir = WORKFLOW_DIR.as_posix()
+    job.status = JobStatusType.RUNNING
+    db_sync.merge(job)
+    db_sync.commit()
+    try:
+        output_dataset.meta = await process_workflow(
+            workflow=workflow,
+            input_paths=input_paths,
+            output_path=output_path,
+            input_metadata=input_dataset.meta,
+            username=username,
+            workflow_dir=WORKFLOW_DIR,
+            logger_name=logger_name,
+            worker_init=worker_init,
+        )
+
+        logger.info(f'END workflow "{workflow.name}"')
+        close_job_logger(logger)
+        db_sync.merge(output_dataset)
 
-    logger.info(f'END workflow "{workflow.name}"')
-    close_job_logger(logger)
-    db.add(output_dataset)
+        job.status = JobStatusType.DONE
+        db_sync.merge(job)
 
-    await db.commit()
+    except TaskExecutionError as e:
+        job.status = JobStatusType.FAILED
+        job.log = (
+            f"TASK ERROR:"
+            f"Task id: {e.workflow_task_id} ({e.task_name}), "
+            f"{e.workflow_task_order=}\n"
+            f"TRACEBACK:\n{str(e)}"
+        )
+        db_sync.merge(job)
```

Comment:
Repeating a previous comment: I think we should call `close_job_logger` also in this `except` branch (i.e. if some task failed).